File: Second Semester(Java and C++)/List 3/C++/Hexagon.cpp
#include "Hexagon.hpp"

#include <cmath>

Hexagon::Hexagon(int side) noexcept(true) {
    this->side = side;
}

double Hexagon::getArea() noexcept(true) {
    return (3 * sqrt(3) * side * side) / 2;
}

double Hexagon::getPerimeter() noexcept(true) {
    return 6 * side;
}

std::string Hexagon::getName() noexcept(true) {
    return "Hexagon";
}

File: Second Semester(Java and C++)/List 3/Java/Task 1/Hexagon.java
public class Hexagon extends Shape {
    private double side;

    public Hexagon(int side) {
        this.side = side;
    }

    @Override
    public double getArea() {
        return (3 * Math.sqrt(3) * Math.pow(side, 2)) / 2;
    }

    @Override
    public double getPerimeter() {
        return 6 * side;
    }

    @Override
    public String getName() {
        return "Hexagon";
    }
}
